File: Practica 3/p1.py
class Pila:
    def __init__(self, n):
        self.arreglo = [None] * n  # Arreglo para almacenar los elementos
        self.top = -1              # Inicialmente, la pila está vacía
        self.n = n                 # Tamaño máximo de la pila

    def push(self, e):
        if self.isFull():
            raise Exception("La pila está llena")
        self.top += 1
        self.arreglo[self.top] = e

    def pop(self):
        if self.isEmpty():
            raise Exception("La pila está vacía")
        elemento = self.arreglo[self.top]
        self.top -= 1
        return elemento

    def peek(self):
        if self.isEmpty():
            raise Exception("La pila está vacía")
        return self.arreglo[self.top]

    def isEmpty(self):
        return self.top == -1

    def isFull(self):
        return self.top == self.n - 1

pila = Pila(5)
pila.push(10)
pila.push(20)
pila.push(30)
print(pila.pop())  # Salida: 30
print(pila.peek()) # Salida: 20
print(pila.isEmpty()) # Salida: False
print(pila.isFull())  # Salida: False

File: Practica 3/p1.java
public class Pila {
    private long[] arreglo;
    private int top;
    private int n;

    public Pila(int n) {
        this.n = n;
        this.arreglo = new long[n];
        this.top = -1;
    }

    public void push(long e) {
        if (isFull()) {
            throw new IllegalStateException("La pila está llena");
        }
        top++;
        arreglo[top] = e;
    }

    public long pop() {
        if (isEmpty()) {
            throw new IllegalStateException("La pila está vacía");
        }
        long elemento = arreglo[top];
        top--;
        return elemento;
    }

    public long peek() {
        if (isEmpty()) {
            throw new IllegalStateException("La pila está vacía");
        }
        return arreglo[top];
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public boolean isFull() {
        return top == n - 1;
    }

    public static void main(String[] args) {
        Pila pila = new Pila(5);
        pila.push(10);
        pila.push(20);
        pila.push(30);
        System.out.println(pila.pop());  // Salida: 30
        System.out.println(pila.peek()); // Salida: 20
        System.out.println(pila.isEmpty()); // Salida: false
        System.out.println(pila.isFull());  // Salida: false
    }
}